extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/EventBusConsumer.java

```diff
@@ -156,7 +161,22 @@ private static void implementInvoke(BeanInfo bean, MethodInfo method, ClassCreat
         ResultHandle messageHandle = invoke.getMethodParam(0);
         ResultHandle result;
 
-        Type paramType = method.parameters().get(0);
+        // Determine if the method is consuming a Message, body, or headers + body.
+        // https://github.com/quarkusio/quarkus/issues/21621
+        Optional<Type> headerType = Optional.empty();
+        Type paramType;
+        if (method.parameters().size() == 2) {
+            Type firstParamType = method.parameters().get(0);
+            if (firstParamType.name().equals(MESSAGE_HEADERS) || firstParamType.name().equals(MUTINY_MESSAGE_HEADERS)) {
+                headerType = Optional.of(firstParamType);
+            } else {
+                // TODO: Throw exception if the 1st param isn't headers,
+                // or assume that the 1st param is the payload?
```

Comment:
Should we validate that a method only has 1 parameter (body) or 2 parameters (header, body)?

Reply:
Nevermind, I see this is handled in `VertxProcessor`.